### PR TITLE
RAC-370: Add IconButton component

### PR DESCRIPTION
**Pull request by ValentinMumble**
**Description (for Contributor and Core Developer)**

- Adding an IconButton component

- Adding Buttons group hierarchy
image

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | Todo
| Added legacy Behats               | Todo
| Added acceptance tests            | Todo
| Added integration tests           | Todo
| Changelog updated                 | Todo
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | Todo
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: akeneo-design-system/src/components/Button/Button.stories.mdx [hunk 1]
```diff
@@ -5,7 +5,7 @@ import {Link} from '../Link/Link.tsx';
 import LinkTo from '@storybook/addon-links/react';
 
 <Meta
-  title="Components/Button"
+  title="Components/Buttons/Button"
```
Comment (StevenVAIDIE) [inline akeneo-design-system/src/components/Button/Button.stories.mdx:8]: The link generate/preview on Introduction does not work anymore

File: akeneo-design-system/src/components/IconButton/IconButton.tsx [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  disabled?: boolean;
+} & React.ButtonHTMLAttributes<HTMLButtonElement>;
+
+/**
+ * The IconButton component is useful to have a clickable icon.
+ */
+const IconButton = React.forwardRef<HTMLButtonElement, IconButtonProps>(
+  ({icon, size = 24, color = 'inherit', ...rest}: IconButtonProps, forwardedRef: Ref<HTMLButtonElement>) => {
+    return (
+      <Container ref={forwardedRef} color={color} {...rest}>
```
Comment (StevenVAIDIE) [inline akeneo-design-system/src/components/IconButton/IconButton.tsx:44]: Missing aria
```suggestion
      <Container aria-disabled={disabled} ref={forwardedRef} color={color} {...rest}>
```

File: akeneo-design-system/src/components/IconButton/IconButton.tsx [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+  /**
+   * The color of the Icon component.
+   */
+  color?: string;
+
+  /**
+   * Whether or not button is disabled.
+   */
+  disabled?: boolean;
```
Comment (StevenVAIDIE) [inline akeneo-design-system/src/components/IconButton/IconButton.tsx:35]: On click is not documented on storybook

```suggestion
  disabled?: boolean;
  
      /**
     * Function called when the user clicks on the button or hit enter when focused.
     */
    onClick?: (event: SyntheticEvent) => void;
```

File: akeneo-design-system/src/components/IconButton/IconButton.tsx [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+  /**
+   * The size of the Icon component.
+   */
+  size?: number;
+
+  /**
+   * The color of the Icon component.
+   */
+  color?: string;
```
Comment (StevenVAIDIE) [inline akeneo-design-system/src/components/IconButton/IconButton.tsx:30]: No level ? 

We cannot restrict allowed colors ?

File: akeneo-design-system/src/components/IconButton/IconButton.tsx [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+type IconButtonProps = {
+  /**
+   * The Icon to display.
+   */
+  icon: ReactElement<IconProps>;
+
+  /**
+   * The size of the Icon component.
+   */
+  size?: number;
```
Comment (StevenVAIDIE) [inline akeneo-design-system/src/components/IconButton/IconButton.tsx:25]: Like button there is no size range ?

File: akeneo-design-system/src/components/IconButton/IconButton.tsx [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+   * Whether or not button is disabled.
+   */
+  disabled?: boolean;
+} & React.ButtonHTMLAttributes<HTMLButtonElement>;
+
+/**
+ * The IconButton component is useful to have a clickable icon.
+ */
+const IconButton = React.forwardRef<HTMLButtonElement, IconButtonProps>(
+  ({icon, size = 24, color = 'inherit', ...rest}: IconButtonProps, forwardedRef: Ref<HTMLButtonElement>) => {
```
Comment (StevenVAIDIE) [inline akeneo-design-system/src/components/IconButton/IconButton.tsx:42]: Maybe add mandatory aria-label and title props ? Or use title value as aria-label ?

File: akeneo-design-system/src/components/Button/Button.stories.mdx (hunk 1 above)
Comment (ValentinMumble) [reply akeneo-design-system/src/components/Button/Button.stories.mdx:8]: Indeed, fixed now

File: akeneo-design-system/src/theme/theme.ts [hunk 7]
```diff
@@ -83,7 +83,7 @@ type Level = 'primary' | 'secondary' | 'tertiary' | 'warning' | 'danger';
 
 const getColor = (color: string, gradient?: number): ((props: AkeneoThemedProps) => string) => ({
   theme,
-}: AkeneoThemedProps): string => theme.color[`${color}${gradient ?? ''}`] as string;
+}: AkeneoThemedProps): string => (theme.color[`${color}${gradient ?? ''}`] ?? color) as string;
```
Comment (StevenVAIDIE) [inline akeneo-design-system/src/theme/theme.ts:86]: I think you can remove this. Not used anymore and this enforce use color from theme

**Review by StevenVAIDIE — APPROVED** [review]

**Review by juliensnz — APPROVED** [review]

File: akeneo-design-system/generator/components/MyComponent/MyComponent.tsx [hunk 8]
```diff
@@ -6,18 +6,18 @@ const MyComponentContainer = styled.div<{level: string}>``;
...
   /**
-   * TODO
+   * TODO.
    */
   children?: ReactNode;
 };
 
 /**
- * TODO
+ * TODO.
```
[the remark this answers is not in the document]
Comment (ValentinMumble) [reply akeneo-design-system/generator/components/MyComponent/MyComponent.tsx:20]: I was always forgetting them 😉

**Pull request merged**